multimodal/src/autogluon/multimodal/utils/metric.py

```diff
@@ -234,17 +234,21 @@ def compute_score(
 
     y = metric_data[Y_TRUE]
     if metric.needs_proba or metric.needs_threshold:
+        y_pred_proba = metric_data[Y_PRED_PROB]
+        y_pred_proba = (
+            y_pred_proba if y_pred_proba.shape[1] > 2 else y_pred_proba[:, pos_label]
+        )  # only use pos_label for binary classification
```

Comment:
What if it is multiclass classification with 2 classes? Is that a valid scenario?

Reply:
That should be binary classification. @cheungdaven correct me if it's wrong: For NER, it's rare but possble to have a multiclass setting of 2 classes if customer is using NER_TOKEN_F1, which is a multiclass F1 score, which does not need proba

Reply:
Instead we can change problem type to binary if there's only 2 classes. Let me check if it's already implemented.